File: validparentheses.cpp
#include <stack>

class Solution {
public:
    bool isValid(string s) {
        stack<char> stk;
        for (size_t i = 0; i < s.size(); i++)
        {
            char c = s.at(i);
            
            if (c == '(' || c == '{' || c == '[')
            {
                stk.push(c);
            }
            else
            {
                if (c == ')')
                {
                    if (!stk.empty() && stk.top() == '(')
                    {
                        stk.pop();
                    }
                    else
                    {
                        return false;
                    }
                }
                if (c == '}')
                {
                    if (!stk.empty() && stk.top() == '{')
                    {
                        stk.pop();
                    }
                    else
                    {
                        return false;
                    }
                }
                if (c == ']')
                {
                    if (!stk.empty() && stk.top() == '[')
                    {
                        stk.pop();
                    }
                    else
                    {
                        return false;
                    }
                }
            }
        }
        if (stk.empty())
        {
            return true;
        }
        return false;
    }
};

File: validParentheses.py
import sys

class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        brMap = {')': '(', '}': '{' , ']': '['}
        for c in s:
            if c in ['(', '{', '[']:
                stk.append(c)
            else:
                if len(stk) != 0 and brMap[c] == stk[-1]:
                    stk.pop()
                else:
                    return False
        if len(stk) == 0:
            return True
        else:
            return False


                    

